qiskit_optimization/algorithms/qrao/magic_rounding.py

```diff
@@ -0,0 +1,448 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Magic basis rounding module"""
+from dataclasses import dataclass
+
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.algorithms.exceptions import AlgorithmError
+from qiskit.primitives import Sampler
+from qiskit.quantum_info import SparsePauliOp
+
+from qiskit_optimization.algorithms import OptimizationResultStatus, SolutionSample
+from .quantum_random_access_encoding import (
+    _z_to_21p_qrac_basis_circuit,
+    _z_to_31p_qrac_basis_circuit,
+)
+from .rounding_common import RoundingContext, RoundingResult, RoundingScheme
+
+
+@dataclass
+class MagicRoundingResult(RoundingResult):
+    """Result of magic rounding."""
+
+    bases: np.ndarray
+    """The bases used for the magic rounding"""
+    basis_shots: np.ndarray
+    """The number of shots used for each basis"""
+    basis_counts: List[Dict[str, int]]
+    """The basis_counts represents the resulting counts obtained by measuring with the bases
+    corresponding to the number of shots specified in basis_shots."""
+
+
+class MagicRounding(RoundingScheme):
+    """Magic rounding scheme
+
+    This scheme is described in https://arxiv.org/abs/2111.03167v2.
```

Comment:
Is there any simple summary that could be added here. I get that the reference is good for more detail but is possible to have something short that captures what this is doing without a user having to read the paper?